mistral-search-engine.ipynb

```diff
@@ -0,0 +1,611 @@
+      ]
+     },
+     "execution_count": 31,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "function_result = tools_to_str(names_to_functions[function_name](**function_params))\n",
+    "function_result"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "\"I found some information about Apple's WWDC 2024. Here are the key points:\\n\\n* The conference took place from June 10 through 14, 2024.\\n* It was an online event, but there was a special in-person experience at Apple Park on opening day for developers and students.\\n* The conference focused on the latest advancements in iOS, iPadOS, macOS, watchOS, tvOS, and visionOS.\\n* iPadOS received new apps like a native calculator, and there were rumors of new Apple Pencil 3 features.\\n* macOS offered AI features, with native apps that leaned on smart tools and generative AI: assisted writing in Pages, a slide deck maker in Keynote, coding in Xcode, all powered by AI.\\n* The event also included video sessions and opportunities to engage with Apple designers and engineers and connect with the worldwide developer community.\\n* Apple supported the next generation of developers through the Swift Student Challenge, with 50 Distinguished Winners being invited to Cupertino for a three-day experience.\\n\\nUnfortunately, I could not find specific details about the 13 biggest announcements. However, based on the information available, it seems that AI and machine learning played a significant role in the conference.\""
+      ]
+     },
+     "execution_count": 32,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "messages.append(ChatMessage(role=\"tool\", name=function_name, content=function_result, tool_call_id=tool_call.id))\n",
+    "\n",
+    "response = client.chat(model=model, messages=messages)\n",
+    "response.choices[0].message.content\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Chat in a chain (cleaner user experience)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "messages = []\n",
+    "\n",
+    "while True:\n",
+    "    input_ = input(\"Ask: \")\n",
+    "    messages.append(ChatMessage(role=\"user\", content=input_))\n",
+    "    response = client.chat(model=model, messages=messages, tools=tools, tool_choice=\"auto\")\n",
```

Comment:
Weirdly it seems it was not updated?

Reply:
just pushed again, and I actually found the cause of the issue so it is fixed now.